lexibank_lieberherrkhobwa.py

```diff
@@ -67,36 +92,39 @@ def cmd_install(self, **kw):
 
             # iterate over the source adding lexemes and collecting cognates
             for row in data[2:]:
+                # Get the language_id, which also allows to skip over
+                # data we don't want
                 lid = slug(row[0])
+
                 if lid in langs:
                     for cid in range(1, len(row), 2):
-                        form = row[cid]
-                        if not form or (form == "NA"):
+                        # Extract the value from the raw data, skipping over
+                        # missing or non-existing forms
+                        value = row[cid].strip()
+                        if not value or value == "NA":
                             continue
-                        ipa = form
-                        ipa = ipa.replace("- ", "-").replace(" ", "_")
-                        ipa = strip_brackets(ipa).strip()
 
-                        if ipa:
-                            # split ipa and remove final/initial underscores if any
-                            # (without regular expressions)
-                            ipa = split_text(ipa, "~/~⪤,")[0]
+                        # Build the form, removing spaces between words
+                        # in the same value and stripping brackets
+                        form = value.replace("- ", "-").replace(" ", "_")
+                        form = strip_brackets(form).strip()
 
-                            if ipa[0] == "_":
-                                ipa = ipa[1:].strip()
-                            if ipa[-1] == "_":
-                                ipa = ipa[:-1].strip()
+                        if form:
+                            # split form and remove final/initial underscores
+                            # if any (without regular expressions)
+                            form = split_text(form, "~/~⪤,")[0]
+                            form = form.strip("_")
 
                             # lingpy now requires this to be an integer
                             cognate_id = cid * 100 + int(row[cid + 1])
 
                             for lex in ds.add_lexemes(
-                                Language_ID=lid,
-                                Parameter_ID=concept_by_index[cid],
-                                Value=form.strip(),
-                                Form=ipa,
-                                Cognacy=cognate_id,
-                                Source=[langs[lid]],
+                                    Language_ID=lid,
+                                    Parameter_ID=concept_by_index[cid],
+                                    Value=value,
+                                    Form=form,
```

Comment:
pylexibank does not accept adding forms explicitly. If you pass them here, they will always be overwritten by the Value. So you do not even need to to that, all you need to do is: handle the form with Dataset.split_values and Dataset.clean_form. Any splitting that is done outside the overridden function won't be passed and doesnt' have any effect. That was my major reason, why I started the discussion last week, since I want to be able in prinicple to pass the form also, as we have datasets with manually annotated forms, and segments. Which is reasonable, as "form" is a concept that you could also do manually. But for the time being, pylexibank does not react on this code.

Reply:
You're right, and this caught an error in the previous code, which was already passing both Value and Form.

Reply:
Just ran `git blame`, it was me who added both Value and Form seven months ago.